modules/trusted-firmware-m/CMakeLists.txt

```diff
@@ -30,6 +30,9 @@ set(TFM_CRYPTO_MODULES
 
 
 if (CONFIG_BUILD_WITH_TFM)
+  if (CONFIG_TFM_SECURE_UART1)
+    list(APPEND TFM_CMAKE_ARGS -DSECURE_UART1=ON)
+  endif()
```

Comment:
Should this have an `else` when secure uart is off, just in case a board has it default on ?

Also, what about boards where this is considered to be an immutable setting, like for the nrf9160 board.

Should the Kconfig reflect this through a forced select when building for that board.
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/c74be3890c9d975976fde1b1a3b2f5742bec34c0/trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf9160/preload.cmake#L9-L18
```
# preload.cmake is used to set things that related to the platform that are both
# immutable and global, which is to say they should apply to any kind of project
# that uses this platform.
...
set(SECURE_UART1 ON)
```

Reply:
> Should this have an else when secure uart is off, just in case a board has it default on ?

I believe so.

> Also, what about boards where this is considered to be an immutable setting, like for the nrf9160 board.

Why is this immutable for 91 ?

Reply:
Ah, I see.

I believe it to be an error to immutably set UART1 to secure.

Reply:
@tejlmand : New revision here: https://github.com/nrfconnect/sdk-zephyr/pull/618/files#diff-dc28570be9d838c291b4be7f1ea2436d81b17f095892a52869cd3cd88317c599R34

I agree with this new revision.

Reply:
> @tejlmand : New revision here: https://github.com/nrfconnect/sdk-zephyr/pull/618/files#diff-dc28570be9d838c291b4be7f1ea2436d81b17f095892a52869cd3cd88317c599R34
> 
> I agree with this new revision.

But that will not enable secure uart for boards where it is default OFF, which is the default case in TF-M.
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/c74be3890c9d975976fde1b1a3b2f5742bec34c0/trusted-firmware-m/config/config_default.cmake#L84
```
set(SECURE_UART1                        OFF         CACHE BOOL      "Enable secure UART1")
```

and loaded from here:
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/c74be3890c9d975976fde1b1a3b2f5742bec34c0/trusted-firmware-m/config/set_config.cmake#L50-L51
```
# Load defaults, setting options not already set
include(config/config_default.cmake)
```

so I believe we need an `if()...else()` to handle both situations.

Reply:
AFAIK all supported boards default to ON.

But I guess it makes sense to future proof it and do an if-else.

Reply:
@tejlmand I do agree that it is a better approach to actually choose both ON and OFF. So I added the else to the statement to handle both cases.

Reply:
From what I can see only the nRF9160, nRF5340, and bl5340 (nRF5340 SoC) defaults to on:
```
$ grep-cmake SECURE_UART
./platform/CMakeLists.txt:        $<$<BOOL:${SECURE_UART1}>:SECURE_UART1>
./platform/ext/target/lairdconnectivity/common/bl5340/config.cmake:set(SECURE_UART1                        ON                   CACHE BOOL      "Enable secure UART1")
./platform/ext/target/lairdconnectivity/common/bl5340/preload.cmake:set(SECURE_UART1 ON)
./platform/ext/target/nordic_nrf/common/nrf9160/config.cmake:set(SECURE_UART1                        ON         CACHE BOOL      "Enable secure UART1")
./platform/ext/target/nordic_nrf/common/nrf9160/preload.cmake:set(SECURE_UART1 ON)
./platform/ext/target/nordic_nrf/common/nrf5340/config.cmake:set(SECURE_UART1                        ON         CACHE BOOL      "Enable secure UART1")
./platform/ext/target/nordic_nrf/common/nrf5340/preload.cmake:set(SECURE_UART1 ON)
./config/config_default.cmake:set(SECURE_UART1                        OFF         CACHE BOOL      "Enable secure UART1")
```
every other board defaults to `OFF`.

Reply:
Sorry, thought this was an NCS PR.

I meant all NCS-supported boards.

Defaulting to OFF globally sounds reasonable.

Reply:
> I believe it to be an error to immutably set UART1 to secure.

Seems the reason the reason is this:
> This removes the stdio_init call when the SECURE_UART1
> is disabled in TF-M. This call will cause a bus fault
> when the SECURE_UART1 is disabled.

which is probably because of the default weak implementation in TF-M:
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/c74be3890c9d975976fde1b1a3b2f5742bec34c0/trusted-firmware-m/platform/ext/common/tfm_platform.c#L16

As TF-M in Zephyr context always will call `stdio_init()`, then I guess allowing users to disable secure uart1 is a recipy for disaster just waiting to happen.

Reply:
Agreed, I was not aware of the platform init file in the downstream TF-M of zephyr.  Closing and it will be changed on downstream ncs.